src/core/modules/NomenclatureItem.cpp

```diff
@@ -302,17 +376,16 @@ void NomenclatureItem::draw(StelCore* core, StelPainter *painter)
 			return;
 	}
 
-	const double screenSize = getAngularSize(core)*M_PI/180.*static_cast<double>(painter->getProjector()->getPixelPerRadAtCenter());
-
-	// We can use ratio of angular size to the FOV to checking visibility of features also!
-	// double scale = getAngularSize(core)/painter->getProjector()->getFov();
-	// if (painter->getProjector()->projectCheck(XYZ, srcPos) && (dist >= XYZ.length()) && (scale>0.04 && scale<0.5))
-
 	// check visibility of feature
 	Vec3d srcPos;
-	if (painter->getProjector()->projectCheck(XYZ, srcPos) && (equPos.length() >= XYZ.length()) && (screenSize>50. && screenSize<750.))
+	const float scale = getAngularSizeRatio(core);
+	NomenclatureItem::NomenclatureItemType niType = getNomenclatureType();
+	if (painter->getProjector()->projectCheck(XYZ, srcPos) && (equPos.length() >= XYZ.length())
+	    && (scale>0.04f && (scale<0.5f || niType>=NomenclatureItem::niSpecialPointPole )))
```

Comment:
Please see whether this sequence of conditions is correct. Maybe a bracket is missing after `scale>0.04f` ?

Reply:
No, this is correct part